src/SourceBuild/content/repo-projects/scenario-tests.proj

```diff
@@ -36,7 +36,7 @@
       <ScenarioTestsResultsDir>$([MSBuild]::NormalizeDirectory('$(ArtifactsTestResultsDir)', 'scenario-tests'))</ScenarioTestsResultsDir>
       <TestXmlOutputPath>$(ScenarioTestsResultsDir)$([System.DateTime]::Now.ToString("yyyy-MM-dd_HH_mm_ss")).xml</TestXmlOutputPath>
       <ScenarioTestsAdditionalArgs>--xml $(TestXmlOutputPath) --target-rid $(TargetRid) --portable-rid $(PortableRid) --no-cleanup --no-traits Category=MultiTFM</ScenarioTestsAdditionalArgs>
-      <ScenarioTestsAdditionalArgs Condition="'$(DotNetBuildPass)' != '2'">$(ScenarioTestsAdditionalArgs) --no-traits Category=RequiresNonTargetRidPackages</ScenarioTestsAdditionalArgs>
+      <ScenarioTestsAdditionalArgs Condition="'$(DotNetBuildPass)' != '2' or '$(DotNetBuildSourceOnly)' == 'true'">$(ScenarioTestsAdditionalArgs) --no-traits Category=RequiresNonTargetRidPackages</ScenarioTestsAdditionalArgs>
```

Comment:
Source-only builds use the MSFT upstream packages and should be able to run the test decorated with that trait, at least with the current state in main. CI tells me that the test is run and that it passes.

Reply:
cc @mthalman

Reply:
It currently works because the workload manifests are downloaded from the Microsoft build and reference the versions built there, but that will change once the manifests are built as part of the sdk repo since they will then use the VMR version numbers.

Reply:
Source-build currently doesn't follow the VMR versioning: https://github.com/dotnet/source-build/issues/4855
Does that make a difference?

FWIW I would prefer to not condition this on source-build until we make the change that requires this to change.

Reply:
> FWIW I would prefer to not condition this on source-build until we make the change that requires this to change.

+1, keep in mind source-build previews unlike UB are shipping. That is to say SB is sensitive to temporary breakages.

Reply:
ok, I still expect this will break but we can deal with it at that point so I'll fold this change into my other PR.